apps/cow-tools/src/app/milkman/(components)/TokenSelect.tsx

```diff
@@ -0,0 +1,210 @@
+import Image from "next/image";
+import { tokenLogoUri } from "public/tokens/logoUri";
+import React, { useEffect, useState } from "react";
+import { formatUnits } from "viem";
+
+import { Dialog } from "#/components/Dialog";
+import Table from "#/components/Table";
+import { useSafeBalances } from "#/hooks/useSafeBalances";
+
+export interface TokenWalletBalance extends TokenInfo {
+  balance: bigint;
+}
+
+export function TokenSelect({
+  onSelectToken,
+  tokenType,
+  selectedToken,
+}: {
+  onSelectToken: (token: TokenWalletBalance) => void;
+  tokenType: "sell" | "buy";
+  selectedToken?: TokenWalletBalance;
+}) {
+  const [open, setOpen] = useState(false);
+  const [token, setToken] = useState<TokenWalletBalance | undefined>(undefined);
+
+  useEffect(() => {
+    if (selectedToken) {
+      setToken(selectedToken);
+    }
+  }, [selectedToken]);
+
+  function handleSelectToken(token: TokenWalletBalance) {
+    onSelectToken(token);
+    setToken(token);
+    setOpen(false);
+  }
+
+  return (
+    <Dialog
+      content={<TokenModal onSelectToken={handleSelectToken} />}
+      isOpen={open}
+      setIsOpen={setOpen}
+    >
+      <div className="flex flex-col">
+        <span className="mb-2 block text-sm text-slate12">
+          Token to {tokenType}
+        </span>
+        <button
+          type="button"
+          className="px-2w-full selection:color-white box-border flex h-[35px] w-full appearance-none items-center justify-between gap-2 rounded-[4px] bg-blue4 px-[10px] py-1 text-[15px] leading-none text-slate12 shadow-[0_0_0_1px] shadow-blue6 outline-none selection:bg-blue9 hover:shadow-[0_0_0_1px_black] focus:shadow-[0_0_0_2px_black] disabled:bg-blue1"
+        >
+          <div className="flex items-center gap-1">
+            <div className="rounded-full bg-white p-[3px]">
+              <Image
+                src={
+                  tokenLogoUri[token?.symbol as keyof typeof tokenLogoUri] ||
+                  "/assets/generic-token-logo.png"
+                }
+                className="rounded-full"
+                alt="Token Logo"
+                height={22}
+                width={22}
+                quality={100}
+              />
+            </div>
+            <div>{token?.symbol}</div>
+          </div>
+          <ChevronDownIcon />
+        </button>
+      </div>
+    </Dialog>
+  );
+}
+
+function TokenModal({
+  onSelectToken,
+}: {
+  onSelectToken: (token: TokenWalletBalance) => void;
+}) {
+  const [tokens, setTokens] = useState<(TokenWalletBalance | undefined)[]>([]);
+  const [tokenSearchQuery, setTokenSearchQuery] = useState("");
+
+  const { assets, loaded } = useSafeBalances();
+
+  useEffect(() => {
+    if (loaded) {
+      const tokens = assets.map((asset) => {
+        return {
+          ...asset.tokenInfo,
+          balance: BigInt(asset.balance),
+        };
+      });
+      setTokens(tokens);
+    }
+  }, [loaded]);
```

Comment:
I think that we might need to filter native tokens here as well. Since they are not erc20.

Reply:
From what I saw it gets every asset, not only erc20. For example GOR

Reply:
Need to filter our ERC20 tokens